commands/uploader.go

```diff
@@ -0,0 +1,155 @@
+package commands
+
+import (
+	"os"
+
+	"github.com/github/git-lfs/lfs"
+)
+
+var uploadMissingErr = "%s does not exist in .git/lfs/objects. Tried %s, which matches %s."
+
+type uploadContext struct {
+	DryRun       bool
+	uploadedOids lfs.StringSet
+}
+
+func newUploadContext(dryRun bool) *uploadContext {
+	return &uploadContext{
+		DryRun:       dryRun,
+		uploadedOids: lfs.NewStringSet(),
+	}
+}
+
+// AddUpload adds the given oid to the set of oids that have been uploaded in
+// the current process.
+func (c *uploadContext) AddUpload(oid string) {
```

Comment:
Nit picky comment: `AddUpload` sounds like a request to add an upload to be done, how about either `AddUploaded` or `SetUploaded` to match `HasUploaded`?